core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -935,12 +957,11 @@ public Void visitValues(ValuesNode node, Void context)
         {
             NodeRepresentation nodeOutput = addNode(node, "Values");
             if (node.getRows().isEmpty()) {
-                for (int i = 0; i < node.getRowCount(); i++) {
-                    nodeOutput.appendDetailsLine("()");
-                }
+                nodeOutput.appendDetailsLine("()");
                 return null;
             }
-            List<String> rows = node.getRows().get().stream()
+            int valuesLimit = Math.min(getExplainValuesLimit(session), node.getRows().get().size());
```

Comment:
We should print a row count, first 9 rows, then `...`, then the last row. We did something similar in `SortedRangeSet#toString`.
Also, avoid truncation in `verbose` mode.